tests/baselines/reference/derivedClassSuperProperties.js

```diff
@@ -0,0 +1,562 @@
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return Derived1;
+}(Base));
+var Derived2 = /** @class */ (function (_super) {
+    __extends(Derived2, _super);
+    function Derived2() {
+        var _this = this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this, _this);
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return Derived2;
+}(Base));
+var Derived3 = /** @class */ (function (_super) {
+    __extends(Derived3, _super);
+    function Derived3() {
+        var _this = this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this);
+        _this = _super.call(this, _this) || this;
+        return _this;
+    }
+    return Derived3;
+}(Base));
+var Derived4 = /** @class */ (function (_super) {
+    __extends(Derived4, _super);
+    function Derived4() {
+        var _this = this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this, _this);
+        _this = _super.call(this, _this) || this;
+        return _this;
+    }
+    return Derived4;
+}(Base));
+var Derived5 = /** @class */ (function (_super) {
+    __extends(Derived5, _super);
+    function Derived5() {
+        var _this = _super.call(this) || this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this);
+        return _this;
+    }
+    return Derived5;
+}(Base));
+var Derived6 = /** @class */ (function (_super) {
+    __extends(Derived6, _super);
+    function Derived6() {
+        var _this = _super.call(this, _this) || this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this);
+        return _this;
+    }
+    return Derived6;
+}(Base));
+var Derived7 = /** @class */ (function (_super) {
+    __extends(Derived7, _super);
+    function Derived7() {
+        var _this = _super.call(this) || this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this, _this);
+        return _this;
+    }
+    return Derived7;
+}(Base));
+var Derived8 = /** @class */ (function (_super) {
+    __extends(Derived8, _super);
+    function Derived8() {
+        var _this = _super.call(this, _this) || this;
+        _this.prop = true;
+        _super.prototype.receivesAnything.call(_this, _this);
+        return _this;
+    }
+    return Derived8;
+}(Base));
+var DerivedWithArrowFunction = /** @class */ (function (_super) {
+    __extends(DerivedWithArrowFunction, _super);
+    function DerivedWithArrowFunction() {
+        var _this = this;
+        _this.prop = true;
+        (function () { return _this; })();
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return DerivedWithArrowFunction;
+}(Base));
+var DerivedWithFunctionDeclaration = /** @class */ (function (_super) {
+    __extends(DerivedWithFunctionDeclaration, _super);
+    function DerivedWithFunctionDeclaration() {
+        var _this = this;

```

Comment:
Transpiled property initializers/parameter properties still access `this` before calling `super()`. There's probably a condition in the TS transform that checks if the first executable statement is `super();`. That's no longer correct

Reply:
Sorry, I'm not sure what change you'd like to see here?

This is already an allowed edge case:

```typescript
class Base { }

class Child extends Base {
    constructor() {
        (() => this)();
        super();
    }
}

new Child()
```

TypeScript gives no checker complaint, and its downpiled output runs without error, but running that directly gives the "Must call super constructor..." error.

Reply:
I don't mean changing the error behavior here. Instead I'm pointing out that the transpiled code is wrong. That's more obvious if you transpile to ES6

```ts
class C extends Object {
  prop = 1;
  constructor() {
    something();
    super();
  }
}

// currently transpiles to something similar to this
class C extends Object {
  constructor() {
    this.prop = 1;  // this gives an error when executed
    return super() || this;
  }
}

// instead it should be
class C extends Object {
  constructor() {
    const _this = super() || this;
    this.prop = 1;
    return _this;
  }
}
```

Reply:
Gotcha, thanks!

This looks to be in `src/compiler/transformers/es2015.ts` within `declareOrCaptureOrReturnThisForConstructorIfNeeded`.